### PR TITLE
style: refactor children handling

### PR DESCRIPTION
**Pull request by emilio**
This way I can slim TNode easily, making easy to share QuerySelector
implementation between Servo and Gecko.

We only use these on elements anyway.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18895)

### CODE REVIEW

File: components/style/invalidation/element/invalidator.rs [hunk 1]
```diff
@@ -500,9 +503,9 @@ where
                 self.invalidate_pseudo_element_or_nac(before, invalidations);
         }
 
-        let node = self.element.as_node();
+        let element = self.element;
         any_descendant |=
-            self.invalidate_dom_descendants_of(node, invalidations);
+            self.invalidate_dom_descendants_of(element, invalidations);
```
Comment (jdm) [inline components/style/invalidation/element/invalidator.rs]: Use `self.element` here and remove the `let element` above.

**Review by jdm — APPROVED** [review]

File: components/style/invalidation/element/invalidator.rs (hunk 1 above)
Comment (emilio) [reply components/style/invalidation/element/invalidator.rs]: I can't, due to the borrow checker and the lack of nll.

Comment (emilio) [reply components/style/invalidation/element/invalidator.rs]: Hmm, maybe I can, but I think I still want to iterate over the DOM children of a doc fragment for querySelector, so I'll refactor the patch a bit.

**Comment by jdm** [comment]
Review status: 0 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[components/layout_thread/dom_wrapper.rs, line 858 at r2](https://reviewable.io:443/reviews/servo/servo/18895#-Kw_WUWbC7SN9jyf5l4T:-Kw_WUWbC7SN9jyf5l4U:b-ozpl0l) ([raw file](https://github.com/servo/servo/blob/f31377304b58a7e8b1df92ec16930b2fc3481fec/components/layout_thread/dom_wrapper.rs#L858)):*
> ```Rust
> impl<'ln> NodeInfo for ServoThreadSafeLayoutNode<'ln> {
>     fn is_element(&self) -> bool {
>         self.pseudo == PseudoElementType::Normal && self.node.is_element()
> ```

The comment still talks about pseudos here; should it be removed?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/18895)*

**Comment by emilio** [comment]
Review status: 0 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[components/layout_thread/dom_wrapper.rs, line 858 at r2](https://reviewable.io:443/reviews/servo/servo/18895#-Kw_WUWbC7SN9jyf5l4T:-Kw_YPi86lHqf6LHRX-9:b-896fix) ([raw file](https://github.com/servo/servo/blob/f31377304b58a7e8b1df92ec16930b2fc3481fec/components/layout_thread/dom_wrapper.rs#L858)):*
<details><summary><i>Previously, jdm (Josh Matthews) wrote…</i></summary><blockquote>

The comment still talks about pseudos here; should it be removed?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/18895)*

**Comment by emilio** [comment]
I bundled this in #18904 to alleviate a bit the queue.

**Pull request closed without merging**